backend/app/models/group.rb

```diff
@@ -0,0 +1,82 @@
+# groups table
+class Group < ActiveRecord::Base
+
+  validates :name,        presence: true
+  validates :description, presence: true
+
+  # Return all groups.
+  #
+  # @return all groups
+  def self.find_all
+    Group.order(:name)
+  end
+
+  # Return all groups beginning with fitst letter l ('*' as non-alphanumeric wildcard).
+  #
+  # @return all groups beginning with fitst letter l ('*' as non-alphanumeric wildcard)
+  def self.find_letter(l)
+    if l == "*"
+      sql = "select * from groups where (name regexp '^[^a-zA-Z0-9].*') order by name"
+    else
+      sql = "select * from groups where name like '#{l}%' order by name"
+    end
+    Group.find_by_sql sql
+  end
+
+  # Return a specific group.
+  #
+  # @param id [Int] the id of the group
+  # @return the groups
+  def self.find_id(id)
+    Group.find_by(id: id)
+  end
+
+  # Create an group
+  #
+  # @param group [Hash] the objet type
+  # @option group[:name] [String] the name
+  # @option group[:description] [String] the description
+  # return the group
+  def self.create(group)
+    # Read the parameters
+    name = Input.text(group[:name])
+    description = Input.text(group[:description])
+
+    group_new = Group.new(
+      name: name,
+      description: description
+    )
+
+    valid = group_new.valid?
+    return false, group_new.errors if !valid
+
+    # Save the group if it is valid
+    group_new.save
+
+    return group_new, false
+  end
+
+  # Update an group
+  #
+  # @param group [Hash] the objet type
```

Comment:
Argh. I tried to scrub all the comments.  I missed this one.

Reply:
Done. Will be in next push.

p.s. I reviewed all the comments in the models (I had focused on the controllers since they are the ones we use to create the API documentation)